src/main/java/com/sigopt/model/Client.java

```diff
@@ -48,10 +48,53 @@ public Experiments(String prefix) {
         }
     }
 
+    public static class ProjectsRoot extends BoundObject {
+        public ProjectsRoot(String prefix) {
+            super(prefix);
+        }
+
+        public PaginatedAPIMethodCaller<Project> list() {
+            return new PaginatedAPIMethodCaller<Project>(
+                "get",
+                this.prefix() + "/projects",
+                Project.class
+            );
+        }
+
+        public APIMethodCaller<Project> create() {
+            return new APIMethodCaller<Project>("post", this.prefix() + "/projects", Project.class);
+        }
+    }
+
+    public static class ProjectsResource extends BoundObject {
```

Comment:
do these need to be separate classes?

Reply:
They don't absolutely need to be, but this prevents `new Client(X).projects(Y).create()`, `new Client(X).projects(Y).list()`, `new Client(X).projects().experiments()` and `new Client(X).projects().fetch()` from creating exceptions at runtime (they will be caught at compile time)